nuxeo-core/nuxeo-core-convert/src/main/java/org/nuxeo/ecm/core/convert/service/ConversionServiceImpl.java

```diff
@@ -247,31 +252,47 @@ public Blob convertBlobToPDF(Blob blob) throws IOException {
                 DownloadService downloadService = Framework.getService(DownloadService.class);
                 blob = replaceURLsByAbsolutePaths(blob, tempDirectory, downloadService::resolveBlobFromDownloadUrl);
                 // Convert the blob to PDF
-                result = convertBlobToMimeType(blob, MimetypeRegistry.PDF_MIMETYPE);
+                result = convertBlobToMimeType(blob, destMimeType);
             } finally {
                 org.apache.commons.io.FileUtils.deleteQuietly(tempDirectory.toFile());
             }
         }
         if (result != null) {
-            adjustPDFBlobName(filename, result);
+            adjustBlobName(filename, result, extension, destMimeType);
         }
         return result;
     }
 
+    @Override
+    public Blob convertBlob(String converterName, BlobHolder blobHolder, String extension, String destMimeType)
+            throws IOException {
+        Blob blob = blobHolder.getBlob();
+        if (converterName == null) {
+            converterName = getConverterName(blob.getMimeType(), destMimeType);
+        }
+        if (converterName != null) {
+            Blob result = convert(converterName, blobHolder, null).getBlob();
```

Comment:
You should give `Collections.emptyMap()` as `convert` method instantiate a `HashMap` if parameters is null.